canvas/services/event.py

```diff
@@ -52,6 +44,29 @@ def get_question_stats(question):
                 answers[answer] = 0
             answers[answer] += 1
 
+    users = []
+    progress_submission_statuses = [0, 0, 0]
+    student_submissions = submissions.filter(uqj__user__role="Student")
+    correct_submissions = student_submissions.filter(is_correct=True).values("uqj__user_id")
+    for correct in correct_submissions:
+        if not users.__contains__(correct["uqj__user_id"]):
+            progress_submission_statuses[0] += 1
+            users.append(correct["uqj__user_id"])
```

Comment:
nit;

What you did works and makes sense but I do think there is a cool way that uses python `set()` to do this better.

If you're curious, look into the `difference()` method of python sets - https://www.geeksforgeeks.org/python-set-difference/ and we can chat about it if what I'm saying sounds like gibberish

Haven't thought about it for too long but I have a hunch this approach saves us some big O complexity and might even save us a database call